File: untitled3/src/main/java/org/example/module/ResolverModule.java
package org.example.module;

import dagger.Module;
import dagger.Provides;
import org.example.entity.Context;
import org.example.entity.Controller;
import org.example.entity.Resolver;

@Module
public class ResolverModule {

    @Provides
    public static Resolver resolver(Context context, Controller controller) {
        return new Resolver(context, controller);
    }

    @Provides
    public static Controller controller(Context context) {
        return new Controller(context);
    }
}

File: untitled3/src/main/java/org/example/module/ResolverModule.kt
package org.example.module

import dagger.Module
import dagger.Provides
import org.example.entity.Context
import org.example.entity.Controller
import org.example.entity.Resolver

@Module
object ResolverModule {
    @JvmStatic
    @Provides
    fun resolver(context: Context, controller: Controller): Resolver {
        return Resolver(context, controller)
    }

    @JvmStatic
    @Provides
    fun controller(context: Context): Controller {
        return Controller(context)
    }
}